lib/spark_api/models/listing.rb

```diff
@@ -235,6 +235,10 @@ def respond_to?(method_symbol, include_all=false)
         end
       end
 
+      def batch_photo_delete(photoIds, args={})
```

Comment:
> Don’t add ! to your destructive (receiver-changing) methods’ names, unless you consider the changing to be “dangerous” and you have a “non-dangerous” equivalent method without the !. If some arbitrary subset of destructive methods end with !, then the whole point of ! gets distorted and diluted, and ! ceases to convey any information whatsoever.

http://davidablack.net/dablog.html#2007/8/15/bang-methods-or-danger-will-rubyist

David A. Black may not be the ultimate authority, but he's pretty consistent on this in The Well Grounded Rubyist also.